clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -1780,6 +1786,15 @@ void CXXNameMangler::mangleDeviceStubName(const IdentifierInfo *II) {
       << II->getName();
 }
 
+void CXXNameMangler::mangleOCLDeviceStubName(const IdentifierInfo *II) {
+  // <source-name> ::= <positive length number> __clang_ocl_kern_imp_
+  // <identifier> <number> ::= [n] <non-negative decimal integer> <identifier>
+  // ::= <unqualified source code identifier>
+  StringRef OCLDeviceStubNamePrefix = "__clang_ocl_kern_imp_";
```

Comment:
I'd suggest using some symbols that a regular user-defined function can't be named to avoid conflict.

Reply:
names starting with double underscore are reserved, and shouldn't conflict with user defined names

Reply:
Well, it's undefined behavior if they do, the same that you can technically define a function in C called `_Z3foov` with a completely different signature from what C++ would expect.  This sort of thing is generally fine as long as we make some effort to avoid unintentional conflicts.